### PR TITLE
Dart deep-equals

### PR DESCRIPTION
**Pull request by noordawod**
The existing implementation of `operator ==` in models uses a simple equality for all properties, which may include Sets, Lists, Iterables, and Maps. But, Dart doesn't perform deep-equality checks on these complex types, so the implementation would always return `false`.

This PR addresses this problem by using the [collection](https://rdr.to/HrcSCSVtQHB) package to perform deep equality.

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [X] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh bin/configs/dart*
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [X] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date
  ``` 
  Commit all changed files.
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@jaumard (2018/09) @josh-burton (2019/12) @amondnet (2019/12) @sbu-WBT (2020/12) @kuhnroyal (2020/12) @agilob (2020/12) @ahmednfwela (2021/08)

### CODE REVIEW

**Comment by noordawod** [comment]
A side note: Flutter uses the `collection` package in its SDK, that's why I don't feel particularly concerned about adding a whole new package just to access one of its functions. But, if any of the maintainers feel a certain unease because of that, I can extract that function and include it in the generated code instead.

**Review by ahmednfwela — APPROVED** [review]
LGTM, I don't care about the extra dependency since it's maintained by the dart team.

File: modules/openapi-generator/src/main/resources/dart2/apilib.mustache [hunk 1]
```diff
@@ -24,14 +24,18 @@ part 'auth/http_bearer_auth.dart';
 {{#models}}{{#model}}part 'model/{{{classFilename}}}.dart';
 {{/model}}{{/models}}
 
+/// A handy comparison utility to deep-compare any arbitrary objects.
+const deepEquality = DeepCollectionEquality();
```
Comment (ahmednfwela) [inline modules/openapi-generator/src/main/resources/dart2/apilib.mustache:28]: is this necessary ?

Comment (noordawod) [reply modules/openapi-generator/src/main/resources/dart2/apilib.mustache:28]: Probably not, given that Dart's `const`s are compiled once. I'll revert.

**Comment by noordawod** [comment]
@ahmednfwela after reverting the last bit, would the PR be ready for merge?

**Review by ahmednfwela — APPROVED** [review]

File: modules/openapi-generator/src/main/resources/dart2/apilib.mustache (hunk 1 above)
Comment (noordawod) [reply modules/openapi-generator/src/main/resources/dart2/apilib.mustache]: ✅

**Pull request merged**